### PR TITLE
fix quest stat trigger and memoryCache hits

### PR DESCRIPTION
**Pull request by Fabio1988**
fix quest_stat trigger

### CODE REVIEW

**Comment by Fabio1988** [comment]
@jfberry do you use `availble_slots` in some queries in DB ?
PMSF and RM did already update to match the column, which was introduced in december :+1:

**Comment by jfberry** [comment]
I do still yes.
Remind me again next week when I get back from holiday and I will push a change ;-)

**Comment by Fabio1988** [comment]
> I do still yes. Remind me again next week when I get back from holiday and I will push a change ;-)

Ok, then I may split those changes in a separate PR. quest_stats should be merged soon, because trigger is wrong :)

File: Sources/RealDeviceMapLib/Misc/MemoryCache.swift [hunk 1]
```diff
@@ -27,7 +27,11 @@ public class MemoryCache<T> {
                     let hit = self.hits[element.key]
                     return hit != nil && now.timeIntervalSince(hit!) < keepTime
                 }
+                let keys = self.store.keys
                 self.lock.unlock()
+                self.hits = self.hits.filter { element in
```
Comment (Fabio1988) [inline Sources/RealDeviceMapLib/Misc/MemoryCache.swift:32]: @123FLO321 what do you think about that change? Some people reported a memory leak to me. especially this hits list would increase in time significantly with pokemon and incident cache

Comment (123FLO321) [reply Sources/RealDeviceMapLib/Misc/MemoryCache.swift:32]: Looks good to me.

**Pull request merged**